clientapi/routing/register.go

```diff
@@ -581,11 +588,39 @@ func handleRegistrationFlow(
 	// TODO: Handle mapping registrationRequest parameters into session parameters
 
 	// TODO: email / msisdn auth types.
+	accessToken, accessTokenErr := auth.ExtractAccessToken(req)
+	if accessTokenErr != nil {
```

Comment:
Hold on a sec. A missing access token will raise an error, except that a missing access token is *only* an error for appservices. /register should work without an access token for non-as based requests

(it's at times like these I wish we had a dedicated endpoint for appservices)